### PR TITLE
[GO]: parametic/ test default propagation style

### PR DESCRIPTION
**Pull request by JianyiGao**
## Description

- Runs `test_headers_precedence_propagationstyle_default_tracecontext_datadog` for `ddtrace-go<1.61.0.dev`.
- Runs`test_headers_precedence_propagationstyle_default_datadog_tracecontext` for `ddtrace-go>=1.62.0`. 
   - `ddtrace v1.61.0` release is currently in progress (rc1 was just released). Changes in https://github.com/DataDog/dd-trace-go/pull/2349 will go out in v1.62.0. 

## Motivation

This PR test default propagation style for golang. Default style changed from `tracecontext,datadog` to `datadog,tracecontext`

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

Once your PR is reviewed, you can merge it! :heart:

## Reviewer checklist

* [x] Check what scenarios are modified. If needed, add the relevant label (`run-parametric-scenario`, `run-profiling-scenario`...). If this PR modifies any system-tests internal, then add the `run-all-scenarios` label ([more info](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md)). 
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] if any of `build-some-image` label is present
  1. is the image labl have been updated ? 
  2. just before merging, locally build and push the image to hub.docker.com
* [ ] if a scenario is added (or removed), add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
There are tests failures in the CI.
I guess we are waiting for merge dd-trace-go. (if it's the case, it would be better to set the PR in Draft status)

**Review by robertomonteromiguel — APPROVED** [review]

**Review by zacharycmontoya — APPROVED** [review]
I'm not familiar with the go changes, but the test skip changes look good to me

File: tests/parametric/test_headers_precedence.py [hunk 1]
```diff
@@ -486,7 +487,6 @@ def test_headers_precedence_propagationstyle_tracecontext(self, test_agent, test
     @missing_feature(context.library == "ruby", reason="library does not yet implement this default configuration")
     @irrelevant(context.library == "cpp", reason="library does not implement this default configuration")
     @irrelevant(context.library == "dotnet", reason="library does not implement this default configuration")
-    @irrelevant(context.library == "golang", reason="library does not implement this default configuration")
     @irrelevant(context.library == "nodejs", reason="library does not implement this default configuration")
```
Comment (mabdinur) [inline tests/parametric/test_headers_precedence.py:490]:
```suggestion
    @missing_feature(context.library < "golang@1.61.0", reason="Default value starting on 1.61.0")
    @irrelevant(context.library == "nodejs", reason="library does not implement this default configuration")
```

File: tests/parametric/test_headers_precedence.py [hunk 2]
```diff
@@ -218,6 +218,7 @@ def test_headers_precedence_propagationstyle_datadog(self, test_agent, test_libr
     @missing_feature(
         context.library == "ruby", reason="Issue: tracecontext,Datadog was never the default configuration"
     )
+    @irrelevant(context.library == "golang", reason="library does not implement this default configuration")
```
Comment (mabdinur) [inline tests/parametric/test_headers_precedence.py:221]:
```suggestion
    @irrelevant(context.library >= "golang@1.61.0", reason="Default value was switched to datadog,tracecontext")
```

File: tests/parametric/test_headers_precedence.py [hunk 3]
```diff
@@ -486,7 +487,7 @@ def test_headers_precedence_propagationstyle_tracecontext(self, test_agent, test
     @missing_feature(context.library == "ruby", reason="library does not yet implement this default configuration")
     @irrelevant(context.library == "cpp", reason="library does not implement this default configuration")
     @irrelevant(context.library == "dotnet", reason="library does not implement this default configuration")
-    @irrelevant(context.library == "golang", reason="library does not implement this default configuration")
+    @irrelevant(context.library < "golang@1.61.0.dev", reason="Default value was updated in v1.61 (w3c phase 2)")
```
Comment (mabdinur) [inline tests/parametric/test_headers_precedence.py:490]:
```suggestion
    @irrelevant(context.library < "golang@1.62.0", reason="Default value was updated in v1.62.0 (w3c phase 2)")
```

File: tests/parametric/test_headers_precedence.py [hunk 4]
```diff
@@ -218,6 +218,7 @@ def test_headers_precedence_propagationstyle_datadog(self, test_agent, test_libr
     @missing_feature(
         context.library == "ruby", reason="Issue: tracecontext,Datadog was never the default configuration"
     )
+    @irrelevant(context.library >= "golang@1.61.0.dev", reason="Default value was switched to datadog,tracecontext")
```
Comment (mabdinur) [inline tests/parametric/test_headers_precedence.py:221]:
```suggestion
    @irrelevant(context.library >= "golang@1.62.0", reason="Default value was switched to datadog,tracecontext")
```

**Pull request merged**